sway/tree/view.c

```diff
@@ -683,6 +683,37 @@ void view_child_destroy(struct sway_view_child *child) {
 	}
 }
 
+struct sway_view *view_from_wlr_surface(struct wlr_surface *wlr_surface) {
+	if (wlr_surface_is_xdg_surface(wlr_surface)) {
+		struct wlr_xdg_surface *xdg_surface =
+			wlr_xdg_surface_from_wlr_surface(wlr_surface);
+		return view_from_wlr_xdg_surface(xdg_surface);
+	}
+	if (wlr_surface_is_xdg_surface_v6(wlr_surface)) {
+		struct wlr_xdg_surface_v6 *xdg_surface_v6 =
+			wlr_xdg_surface_v6_from_wlr_surface(wlr_surface);
+		return view_from_wlr_xdg_surface_v6(xdg_surface_v6);
+	}
+	if (wlr_surface_is_xwayland_surface(wlr_surface)) {
+		struct wlr_xwayland_surface *xsurface =
+			wlr_xwayland_surface_from_wlr_surface(wlr_surface);
+		return view_from_wlr_xwayland_surface(xsurface);
+	}
+	if (wlr_surface_is_subsurface(wlr_surface)) {
+		struct wlr_subsurface *subsurface =
+			wlr_subsurface_from_wlr_surface(wlr_surface);
+		return view_from_wlr_surface(subsurface->parent);
+	}
+	if (wlr_surface_is_layer_surface(wlr_surface)) {
+		wlr_log(L_DEBUG, "Surface of type layer has no view");
```

Comment:
I kind of want to know if someone tries to use idle inhibitor on something that doesn't have a view, but I can log it on that side; will remove this.

Reply:
My fear is that this'll be used in something frequent like testing which view is under a pointer or something, then get logged every frame as you drag the pointer around the screen.

Reply:
I plan on using it once at idle inhibitor init rather than every frame (even if there will be a check at every transaction which will be somewhat redundant on interactive resize :/), but yeah can't know where this will be used later. It's gone (force pushed)